internal/handlers/pg/pgdb/settings.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pgdb
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+
+	"github.com/jackc/pgx/v4"
+	"golang.org/x/exp/slices"
+
+	"github.com/FerretDB/FerretDB/internal/fjson"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+const (
+	// Internal collections prefix.
+	collectionPrefix = "_ferretdb_"
+
+	// PostgreSQL max table name length.
+	maxTableNameLength = 63
+)
+
+// CreateSettingsTable creates FerretDB settings table if it doesn't exist.
+func (pgPool *Pool) CreateSettingsTable(ctx context.Context, db string) error {
+	tables, err := pgPool.tables(ctx, db)
```

Comment:
Another thing that needs a transaction? Same case: we have two queries that query tables and couldn't find `collectionPrefix+"settings"`. Will both of them try to create it?